packages/backend-defaults/src/entrypoints/cache/CacheManager.ts

```diff
@@ -191,61 +342,96 @@ export class CacheManager {
     defaultTtl?: number,
     storeOptions?: CacheStoreOptions,
   ) {
-    if (!this.storeFactories.hasOwnProperty(store)) {
-      throw new Error(`Unknown cache store: ${store}`);
-    }
     this.logger = logger;
-    this.store = store as keyof CacheManager['storeFactories'];
+    this.store = store;
     this.connection = connectionString;
     this.errorHandler = errorHandler;
     this.defaultTtl = defaultTtl;
     this.storeOptions = storeOptions;
+
+    this.storeFactories = {
+      redis: this.createRedisStoreFactory(),
+      valkey: this.createValkeyStoreFactory(),
+      memcache: this.createMemcacheStoreFactory(),
+      memory: this.createMemoryStoreFactory(),
```

Comment:
This is odd too - the set of factories was always intended to be a static set, and then you call out to them if they are configured. Having _a factory_ in place should not have any side effects, CALLING the factory should have side effects. I hope you can refactor this to not change around how the very core of everything works